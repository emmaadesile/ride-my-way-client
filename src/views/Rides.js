import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import fetchRides from "../actions/rides";
import Navbar from "../components/common/Navbar";
import Loading from "../components/Loading/Loading";
import ViewRide from "../components/ViewRide";
import car from "../assets/img/car.png";

class Rides extends Component {
  state = {
    showModal: false,
    rideid: ''
  };

  componentDidMount = () => {
    const { fetchAllRides } = this.props;
    fetchAllRides();
  };

  showModal = (event) => {
    const { rideid } = event.target.dataset;
    this.setState({
      showModal: true,
      rideid
    });
  };

  closeModal = () => {
    this.setState({
      showModal: false
    });
  };


  render() {
    const { rides, loading } = this.props;
    const { showModal, rideid } = this.state;

    return (
      <Fragment>
        <div className="wrapper3">
          <Navbar />
          {loading ? (
            <Loading />
          ) : (
            <section className="rides mt-3 mb-2 pb-4">

              <h4 className="ride-heading ">{rides.length === 0 ? 'No rides available' : 'Available Rides'}</h4>
              {rides.map(ride => (
                <div className="ride ride-1" key={ride.ride_id}>
                  <div className="icon">
                    <img src={car} alt="car" />
                  </div>
                  <h5 className="ride-info">
                    {ride.location} to {ride.destination}
                  </h5>
                  <button
                    id="btn-modal"
                    className="btn btn__secondary"
                    onClick={this.showModal}
                    data-rideid={ride.ride_id}
                  >
                    View
                  </button>
                </div>
              ))}
            </section>
          )}
          {showModal ? <ViewRide rideid={rideid} closeModal={this.closeModal} /> : null}
        </div>
      </Fragment>
    );
  }
}

const mapStateToProps = state => {
  const { rides } = state;
  return {
    rides: rides.data,
    loading: rides.loading,
    error: rides.error
  };
};

const mapDispatchToProps = dispatch => ({
  fetchAllRides() {
    return dispatch(fetchRides());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Rides);
