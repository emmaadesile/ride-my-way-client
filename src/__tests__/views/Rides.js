import React from "react";
import { shallow } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import toJson from 'enzyme-to-json';
import Rides from "../../views/Rides.jsx";

const mockStore = configureMockStore();
const store = mockStore();

const mockFunction = param => jest.fn(param);

describe("Rides View", () => {
  test("renders the rides page", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Rides fetchRides={mockFunction} />
      </Provider>
    );
    expect(wrapper.exists()).toBe(true);
  });

  it('should render correctly', () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Rides />
      </Provider>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
