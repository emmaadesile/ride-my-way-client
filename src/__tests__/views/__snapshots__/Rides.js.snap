// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rides View should render correctly 1`] = `<Connect(Rides) />`;
